### PR TITLE
[cli] Skip authentication if auth server is not available, improve workflow error reporting formatting

### PR DESCRIPTION
**Pull request by Kami**
This pull request includes 3 fixes / improvements:

1. If auth server is unavailable when we try to authenticate we now assume auth is disabled and still try to proceed with the request to the API service without a token. If auth is indeed enabled, request to the API service will fail. Previously, if credentials were specified in `~/.st2/config` and the auth service was unavailable (disabled) CLI would explode.

2. Improve workflow error formatting - there were some edge cases with Mistral errors where CLI would explode and not report the error correctly.2

3. Update tests so they don't throw if the CLI config file is present.

### CODE REVIEW

File: st2client/tests/base.py [hunk 1]
```diff
@@ -86,6 +86,8 @@ def setUp(self):
             if var in os.environ:
                 del os.environ[var]
 
+        os.environ['ST2_SKIP_CLI_CONFIG'] = '1'
```
Comment (lakshmi-kannan) [inline st2client/tests/base.py]: Is this needed only for tests? I am not seeing a genuine use case outside tests?

**Comment by lakshmi-kannan** [comment]
LGTM.

File: st2client/st2client/shell.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
 import traceback
 
 import six
+import requests
```
Comment (lakshmi-kannan) [inline st2client/st2client/shell.py]: Nit: alpha order.

File: st2client/tests/base.py (hunk 1 above)
Comment (Kami) [inline st2client/tests/base.py]: Heh, that's actually kinda related to the auth change we talked about.

In short - there are scenarios where you want don't want to use the CLI config (e.g. you only want to use values provided as the command line arguments).

Without this option, you need to "temporary" remove or move the config.

Comment (lakshmi-kannan) [inline st2client/tests/base.py]: >Without this option, you need to "temporary" remove or move the config.
Yeah, that's more explicit and will be the thing with least surprises. People will forget to reset env variables.

Comment (Kami) [inline st2client/tests/base.py]: Environment variables can also be specified on per command invocation basis. In fact, that's how I envisioned people would use it.

An alternative would be adding ``--skip-config`` flag. I'm fine with that.

Comment (lakshmi-kannan) [inline st2client/tests/base.py]: I'd leave the env variable thing undocumented for now. I am pretty sure people will abuse it by using `export` and will complain on IRC. I do like --skip-config.

Comment (Kami) [inline st2client/tests/base.py]: fd2130b6257740844867db3ea5939b705847079b
